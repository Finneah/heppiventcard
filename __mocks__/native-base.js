jest.mock('native-base', () => {
  return {
    StyleProvider: jest.fn().mockImplementation((x) => x),
  };
});
