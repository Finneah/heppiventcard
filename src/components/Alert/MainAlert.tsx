import {Alert, Box} from 'native-base';

import * as React from 'react';

type Props = {
  status: string;
  title: string;
  description: string;
  showAlert: boolean;
};

export const MainAlert: React.FC<Props> = ({
  status,
  title,
  description,
  showAlert = false,
}) => {
  return showAlert ? (
    <Box safeArea>
      <Alert status={status} w="100%">
        <Alert.Icon />
        <Box flex={1} ml={2}>
          <Alert.Title>{title}</Alert.Title>
          <Alert.Description>{description}</Alert.Description>
        </Box>
      </Alert>
    </Box>
  ) : null;
};
