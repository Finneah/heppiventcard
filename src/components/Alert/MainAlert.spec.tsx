import 'react-native';
import * as React from 'react';
import {render} from '@testing-library/react-native';

// Note: test renderer must be required after react-native.

import {NativeBaseProvider} from 'native-base';

import MainAlert from './MainAlert';

const wrapper = ({children}: any) => (
  <NativeBaseProvider
    initialWindowMetrics={{
      frame: {x: 0, y: 0, width: 0, height: 0},
      insets: {top: 0, left: 0, right: 0, bottom: 0},
    }}>
    {children}
  </NativeBaseProvider>
);

describe('MainAlert', () => {
  test('renders correctly', () => {
    const mainAlert = render(
      <MainAlert
        status={'error'}
        title={'title'}
        description={'description'}
        showAlert={true}
      />,
      {
        wrapper,
      },
    );

    expect(mainAlert.toJSON()).toMatchSnapshot();
  });
});
