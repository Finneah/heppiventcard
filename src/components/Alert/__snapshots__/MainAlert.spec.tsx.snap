// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainAlert renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <View
      alignItems="center"
      bg="#fee2e2"
      borderRadius="md"
      flexDirection="row"
      justifyContent="flex-start"
      p={3}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fee2e2",
            "borderRadius": 4,
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "paddingBottom": 12,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 12,
            "width": "100%",
          },
        ]
      }
      w="100%"
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <RNSVGSvgView
          accessibilityRole="image"
          align="xMidYMid"
          bbHeight={24}
          bbWidth={24}
          color={4293870660}
          focusable={false}
          meetOrSlice={0}
          minX={0}
          minY={0}
          size={6}
          stroke=""
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Array [
                Object {
                  "color": "#ef4444",
                  "height": 24,
                  "width": 24,
                },
              ],
              Object {
                "flex": 0,
                "height": 24,
                "width": 24,
              },
            ]
          }
          tintColor={4293870660}
          vbHeight={24}
          vbWidth={24}
        >
          <RNSVGGroup
            propList={
              Array [
                "stroke",
              ]
            }
            stroke={null}
          >
            <RNSVGGroup>
              <RNSVGPath
                d="M23.119,20,13.772,2.15h0a2,2,0,0,0-3.543,0L.881,20a2,2,0,0,0,1.772,2.928H21.347A2,2,0,0,0,23.119,20ZM11,8.423a1,1,0,0,1,2,0v6a1,1,0,1,1-2,0Zm1.05,11.51h-.028a1.528,1.528,0,0,1-1.522-1.47,1.476,1.476,0,0,1,1.448-1.53h.028A1.527,1.527,0,0,1,13.5,18.4,1.475,1.475,0,0,1,12.05,19.933Z"
                fill={
                  Array [
                    2,
                  ]
                }
                propList={
                  Array [
                    "fill",
                    "stroke",
                  ]
                }
                stroke={null}
              />
            </RNSVGGroup>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <View
        style={
          Array [
            Object {
              "width": 12,
            },
          ]
        }
        width={3}
      />
      <View
        flex={1}
        ml={2}
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginLeft": 8,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            color="muted.700"
            fontFamily="body"
            fontSize="md"
            fontStyle="normal"
            fontWeight="bold"
            lineHeight={5}
            style={
              Array [
                Object {
                  "color": "#404040",
                  "fontSize": 16,
                  "fontStyle": "normal",
                  "fontWeight": "700",
                  "lineHeight": 20,
                },
              ]
            }
          >
            title
          </Text>
        </View>
        <View
          flexShrink={1}
          style={
            Array [
              Object {
                "flexShrink": 1,
              },
            ]
          }
        >
          <Text
            color="muted.700"
            fontFamily="body"
            fontSize="sm"
            fontStyle="normal"
            fontWeight={400}
            style={
              Array [
                Object {
                  "color": "#404040",
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                },
              ]
            }
          >
            description
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
